loopy/type_inference.py

```diff
@@ -593,6 +593,12 @@ def map_sub_array_ref(self, expr):
 
     map_fortran_division = map_quotient
 
+    def map_nan(self, expr):
+        if expr.data_type is None:
+            return [NumpyType(np.dtype(np.float64))]
```

Comment:
To be consistent with the handling here:

https://github.com/inducer/loopy/blob/b04bf37037a7f17c3725b8543378d972d0485ab4/loopy/type_inference.py#L376-L379

I'd say this should be `float32`.

Reply:
TIL. Fixed, thanks!